src/Eloquent/Concerns/InteractsWithRelations.php

```diff
@@ -270,7 +292,15 @@ public function connectBelongsToManyRelation(Relations\BelongsToMany $relation,
                 $pivotInstance::unguard();
             }
 
-            $relation->sync($data, $detaching);
+            $results = $relation->sync($data, $detaching);
+
+            foreach ($relation->getRelated()->findMany($results['attached']) as $model) {
```

Comment:
On all the other relations we do the authorization first and then set the relation. Any reason it's the other way around here?

Reply:
`sync` gives back the ids of the attached and detached models. Otherwise we'd have to calculate the differences manually first.

Reply:
No wait, you're right. We should check that before syncing.